x-pack/plugins/search_indices/public/components/indices/details_page.tsx

```diff
@@ -208,20 +200,34 @@ export const SearchIndexDetailsPage = () => {
             data-test-subj="searchIndexDetailsHeader"
             pageTitle={index?.name}
             rightSideItems={[
-              <EuiFlexGroup>
+              <EuiFlexGroup gutterSize="none">
                 <EuiFlexItem>
                   <EuiButtonEmpty
-                    href={docLinks.links.apiReference}
-                    target="_blank"
-                    iconType="documentation"
-                    data-test-subj="ApiReferenceDoc"
+                    href={!isDocumentsExists ? docLinks.links.apiReference : playgroundUrl}
+                    target={!isDocumentsExists ? '_blank' : undefined}
+                    isLoading={isDocumentsLoading}
+                    iconType={!isDocumentsExists ? 'documentation' : 'launch'}
+                    data-test-subj={!isDocumentsExists ? 'ApiReferenceDoc' : 'useInPlaygroundLink'}
                   >
-                    {i18n.translate('xpack.searchIndices.indexActionsMenu.apiReference.docLink', {
-                      defaultMessage: 'API Reference',
-                    })}
+                    <FormattedMessage
+                      id="xpack.searchIndices.indexAction.buttonLabel"
+                      defaultMessage="{buttonLabel}"
+                      values={{
+                        buttonLabel: isDocumentsLoading
+                          ? 'Loading'
+                          : !isDocumentsExists
+                          ? 'API Reference'
```

Comment:
Needs localisation. I wonder if the code would read better if it was one conditional (has documents) returning the Playground button / API Reference button vs many small conditionals overriding the design for one button.